### PR TITLE
Fill-in comment author data from the provided WP_User when possible

### LINKED ISSUE #2632: post comment as logged in user can't fill/get right author info
Auth by cookie(on the site
POST /wp/v2/comments with data 

``` json
{
  "author": 1,
  "author_name": "Epix",
  "author_url": "",
  "content": "test",
  "post": "6"
}
```

Post is successful and get comment id 42. Name and avatar is correct in dashboard - comments
GET  /wp/v2/comments/42

``` json
{
  "id": 42,
  "post": 6,
  "parent": 0,
  "author": 1,
  "author_name": "Epix",
  "author_url": "",
  "content": {
    "rendered": "<p>test</p>\n"
  },
  "author_avatar_urls": {
    "24": "http://0.gravatar.com/avatar/?s=24&d=mm&r=g",
    "48": "http://0.gravatar.com/avatar/?s=48&d=mm&r=g",
    "96": "http://0.gravatar.com/avatar/?s=96&d=mm&r=g"
  },
...
}
```

Note that the author_avatar_urls are wrong.

I didn't post data with author_email as think it's not supposed to. Author id should provide enough information for author info of this comment. The author_name and other info should be filled automatically when posting the comment or retrieving the comment.

### PR DESCRIPTION
**Pull request by rachelbaker**
If only the comment author UD (for a known author) is provided when creating a comment, the author's email and URL are not be properly filled in; ID should be sufficient to retrieve supporting author information. This patch uses the ID to retrieve the author's display name, email and URL when preparing a comment for the database.
- [x] Needs description
- [x] Needs tests

Fixes #2632

### CODE REVIEW

**Comment by rachelbaker** [comment]
@kadamwhite just needs tests and resolved merge conflicts.

**Comment by rachelbaker** [comment]
@kadamwhite added one test, needs some more for when an `author` value is passed, how the author data is populated.  👶 woke up, but I will try to add more over the weekend.

File: lib/endpoints/class-wp-rest-comments-controller.php [hunk 1]
```diff
@@ -897,7 +909,6 @@ public function get_item_schema() {
 					'context'      => array( 'view', 'edit', 'embed' ),
 					'arg_options'  => array(
 						'sanitize_callback' => 'sanitize_text_field',
-						'default'           => '',
```
Comment (kadamwhite) [inline lib/endpoints/class-wp-rest-comments-controller.php:900]: :+1:

File: tests/test-rest-comments-controller.php [hunk 2]
```diff
@@ -1152,6 +1160,46 @@ public function test_create_comment_require_login() {
 		$this->assertEquals( 'rest_comment_login_required', $data['code'] );
 	}
 
+	public function test_create_item_invalid_author() {
+		wp_set_current_user( $this->admin_id );
+
+		$params = array(
+			'post'         => $this->post_id,
+			'author'       => REST_TESTS_IMPOSSIBLY_HIGH_NUMBER,
+			'content'      => "It\'s all over\, people! We don\'t have a prayer!",
```
Comment (kadamwhite) [inline tests/test-rest-comments-controller.php]: Doesn't double-quoting obviate the need to escape the single quotes?

**Comment by rachelbaker** [comment]
@kadamwhite #reviewmerge please

File: tests/test-rest-comments-controller.php (hunk 2 above)
Comment (rachelbaker) [reply tests/test-rest-comments-controller.php]: @kadamwhite you are right. fixed in e25d95b

**Review by kadamwhite — APPROVED** [review]
🚀

**Pull request merged**